### PR TITLE
Add linters for ROS1

### PR DESCRIPTION
**Pull request by ojasjoshi**
*Issue #, if available:*

*Description of changes:*
- Added linters for ROS1
- Fixed linting errors.
- Added .py extensions to rospy nodes. Changed their references in the required launch files.

*Tests*
- Tested locally on Melodic+G9 that the build succeeds and the all launch files work as expected.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ahtsan** [comment]
Not relevant to this PR, but just realized `cloudwatch_simulation` is missing `setup.py`

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  ament_lint:
+    runs-on: ubuntu-latest
+    container:
+      image: rostooling/setup-ros-docker:ubuntu-bionic-ros-eloquent-ros-base-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        linter: [copyright, flake8, pep257, xmllint]
+    steps:
+      - run: sudo chown -R rosbuild:rosbuild "$HOME" .
```
Comment (mm318) [inline .github/workflows/lint.yml]: We shouldn't need this anymore since https://github.com/ros-tooling/setup-ros-docker/issues/7 has been resolved.

**Comment by mm318** [comment]
Similar to the problem mentioned at https://github.com/aws-robotics/aws-robomaker-sample-application-helloworld/pull/98#issuecomment-755786594, I think many files (like [these](https://github.com/aws-robotics/aws-robomaker-sample-application-cloudwatch/tree/ros1/robot_ws/src/cloudwatch_robot/nodes)) aren't being linted because they don't have the `.py` extension even though they are Python scripts.

File: robot_ws/src/cloudwatch_robot/nodes/monitor_distance_to_goal.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            value=distance,
+            time_stamp=rospy.Time.from_sec(time.time()),
+            dimensions=dimensions,
+        )
+
+        self.metrics_pub.publish(MetricList([metric]))
+
+
+def main():
+    rospy.init_node('monitor_goal_to_distance')
```
Comment (mm318) [inline robot_ws/src/cloudwatch_robot/nodes/monitor_distance_to_goal.py:70]: There is a line (`monitor = MonitorDistanceToGoal()`) missing here.

File: robot_ws/src/cloudwatch_robot/nodes/monitor_obstacle_distance.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            dimensions=dimensions,
+        )
+
+        self.metrics_pub.publish(MetricList([metric]))
+
+
+def main():
+    rospy.init_node('monitor_obstacle_distance')
+    try:
+        rospy.spin()
```
Comment (mm318) [inline robot_ws/src/cloudwatch_robot/nodes/monitor_obstacle_distance.py]: There is a line (`monitor = MonitorDistanceToGoal()`) missing here.

File: robot_ws/src/cloudwatch_robot/nodes/monitor_speed.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        time_stamp=rospy.Time.from_sec(time.time()),
+        dimensions=dimensions,
+    )
+
+    return MetricList([linear_speed, angular_speed])
+
+
+def main():
+    rospy.init_node('speed_monitor')
+    rospy.spin()
```
Comment (mm318) [inline robot_ws/src/cloudwatch_robot/nodes/monitor_speed.py]: There's a line of code:
```
    monitor = Monitor(data_topic="/odom",
                      data_msg=Odometry,
                      metric_topic="/metrics",
                      transform=odom_to_speed)
```
missing here.

File: simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager [hunk 5]
```diff
@@ -1,305 +0,0 @@
...
-import numpy as np
-import time
-
-
-class GoalGenerator():
-    '''
-    Reads map data published on /map and /map_metadata topics and provides valid random goal poses in the map.
-
-    Assumes that the map is held static after node initialisation and is not updated while the node is running.
-    '''
```
Comment (mm318) [inline simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager:35]: I think we should still keep these comments, maybe in some format like
```
# Reads map data published on /map and /map_metadata topics and provides valid random goal poses in the map.
#
# Assumes that the map is held static after node initialisation and is not updated while the node is running.
class GoalGenerator:
...
```

File: simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager [hunk 6]
```diff
@@ -1,305 +0,0 @@
...
-            }
-
-        return p
-
-
-    def check_noise(self, x, y, row_id=None):
-        '''
-        - description:
-            Low resolution/ noisy sensor data might lead to noisy patches in the map.
-            This function checks if the random valid point is not a noisy bleap on the map by looking for its neighbor consistency.
```
Comment (mm318) [inline simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager:132]: Let's keep the more detailed comments. Reformat it as necessary.

File: simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager.py [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+                    return False
+
+        return True
+
+    # def __iter__(self):
+    #     return self
+
+    # def next(self):
+    #     """For python 2.x support."""
+    #     return self.__next__()
```
Comment (mm318) [inline simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager.py]: Remove commented out code.

File: simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager [hunk 8]
```diff
@@ -1,305 +0,0 @@
...
-                - pose: 
-                      position: 
-                        x: -5.41667556763
-                        y: -3.14395284653
-                        z: 0.0
-                      orientation: 
-                        x: 0.0
-                        y: 0.0
-                        z: 0.785181432231
-                        w: 0.619265789851
```
Comment (mm318) [inline simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager:222]: Let's keep the more detailed comments.

**Review by mm318 — CHANGES_REQUESTED** [review]

**Comment by mm318** [comment]
> Not relevant to this PR, but just realized `cloudwatch_simulation` is missing `setup.py`

What are the implications of that? Is it because it doesn't contain a ROS node?

File: robot_ws/src/cloudwatch_robot/nodes/monitor_distance_to_goal.py (hunk 2 above)
Comment (ojasjoshi) [reply robot_ws/src/cloudwatch_robot/nodes/monitor_distance_to_goal.py:70]: Oh good catch! The linter throws a error here "F841 local variable 'monitor' is assigned to but never used" but this node is non-functional without the object initialisation.

File: simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager (hunk 8 above)
Comment (ojasjoshi) [reply simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager:222]: Not sure what that comment was trying to describe exactly.

File: simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager (hunk 6 above)
Comment (ojasjoshi) [reply simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager:132]: I believe the concise description is good enough. But I can add the previous one in the description.

File: simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager.py [hunk 9]
```diff
@@ -0,0 +1,371 @@
...
+                    'Route mode is %s, getting next goal',
+                    self.route_mode)
+                try:
+                    if self.route_mode == 'dynamic':
+                        # TODO: flake8 linting standard does not allow over-riding
+                        # the built-in python method next() which is required to
+                        # make 'self.goals' a iterator.
+                        current_goal = self.to_move_goal(self.goals.get_next())
+                    else:
+                        current_goal = self.to_move_goal(next(self.goals))
```
Comment (mm318) [inline simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager.py:328]: Are the two branches any different functionally?

**Review by mm318 — CHANGES_REQUESTED** [review]
Please remove the `.DS_Store` files.

File: simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager.py (hunk 9 above)
Comment (mm318) [reply simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager.py:328]: Okay, I understand now that `goals` is a different type in different `route_mode`s.

**Review by mm318 — APPROVED** [review]

**Pull request merged**